src/Core/src/Container.php

```diff
@@ -153,13 +153,15 @@ public function has(string $id): bool
     /**
      * Make a Binder proxy to configure bindings for a specific scope.
      *
-     * @param null|string $scope Scope name.
+     * @param null|\BackedEnum|string $scope Scope name.
      *        If {@see null}, binder for the current working scope will be returned.
      *        If {@see string}, the default binder for the given scope will be returned. Default bindings won't affect
      *        already created Container instances except the case with the root one.
      */
-    public function getBinder(?string $scope = null): BinderInterface
+    public function getBinder(string|\BackedEnum|null $scope = null): BinderInterface
     {
+        $scope = $scope instanceof \BackedEnum ? $scope->value : $scope;
```

Comment:
Check like `is_object` or `is_enum` might be faster than `instanceof`

Reply:
Changed to is_object